PyPortal_Wakeup_Light/wake_up_light.py

```diff
@@ -0,0 +1,198 @@
+
+# set neopixel min and max brightness
+BRIGHTNESS = 0
+MIN_BRIGHTNESS = 0
+MAX_BRIGHTNESS = 0.85
+#initialize neopixel strip
+num_pixels = 30
+ORDER = neopixel.RGBW
+strip = neopixel.NeoPixel(board.D3, num_pixels, brightness=BRIGHTNESS,
+                          pixel_order=ORDER)
+strip.fill(0)
+# color of strip
+WHITE = (255, 255, 255, 255)
+# number of minutes it takes for strip to fade from min to max
+light_minutes = 30
+
+# determine the current working directory
+# needed so we know where to find files
+cwd = ("/"+__file__).rsplit('/', 1)[0]
+
+# Initialize the pyportal object and let us know what data to fetch and where
+# to display it
+pyportal = PyPortal(status_neopixel=board.NEOPIXEL,
+                    default_bg=0x000000)
+
+big_font = bitmap_font.load_font(cwd+"/fonts/Nunito-Light-75.bdf")
+big_font.load_glyphs(b'0123456789:AP') # pre-load glyphs for fast printing
+print('loading fonts...')
+info_font = bitmap_font.load_font(cwd+"/fonts/Nunito-Black-17.bdf")
+info_font.load_glyphs(b'0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-,.:/ ')
+
+time_color = 0xFFFFFF
+time_position = (75,130)
+time_textarea = Label(big_font, max_glyphs=15, color=time_color,
+                      x=time_position[0], y=time_position[1])
+
+wakeup_time_color = 0xFFFFFF
+wakeup_time_position = (15,200)
+wakeup_time_textarea = Label(info_font, max_glyphs=30, color=wakeup_time_color,
+                             x=wakeup_time_position[0], y=wakeup_time_position[1])
+
+light_on_time_color = 0xFFFFFF
+light_on_time_position = (15,220)
+light_on_time_textarea = Label(info_font, max_glyphs=30, color=light_on_time_color,
+                               x=light_on_time_position[0], y=light_on_time_position[1])
+
+pyportal.splash.append(time_textarea)
+pyportal.splash.append(wakeup_time_textarea)
+pyportal.splash.append(light_on_time_textarea)
+
+def whichDay():
+    now = time.localtime()
+    current_hour, current_minutes = now[3:5]
+    current_day = now[6]
+    wake_up_hour, wake_up_minutes = default_wake_up.split(":")
+    wake_up_hour = int(wake_up_hour)
+    wake_up_minutes = int(wake_up_minutes[:-1])
+    print(wake_up_hour, ":", wake_up_minutes)
+     # if it's after midnight and before the default wakeup time, display the wake up time of today
+    for day in range(len(wake_up_times)):
+        if current_hour < wake_up_hour and current_minutes < wake_up_minutes:
+            if current_day == day:
+                input_wake_up_time = wake_up_times[day]
+    # set wake up time to the next day's wake up time the night before
+        else:
+            if current_day == 6:
+                input_wake_up_time = wake_up_times[0]
+            else:
+                if current_day == day:
+                    input_wake_up_time = wake_up_times[day+1]
+    input_wake_up_time_text = "Wake up at " + input_wake_up_time
+    wakeup_time_textarea.text = input_wake_up_time_text
+    return input_wake_up_time
+
+def subtract30min(time_before):
```

Comment:
you should be able to do `time` based math instead of this parsing code
https://forum.micropython.org/viewtopic.php?t=959#p5492